tests/test_freqtradebot.py

```diff
@@ -1877,7 +1898,10 @@ def test_handle_overlapping_signals(default_conf, ticker, limit_buy_order_open,
     mocker.patch.multiple(
         'freqtrade.exchange.Exchange',
         fetch_ticker=ticker,
-        buy=MagicMock(return_value=limit_buy_order_open),
+        create_order=MagicMock(side_effect=[
+            limit_buy_order_open,
+            {'id': 1234553382},
+        ]),
```

Comment:
These instances is where having these as seperate methods simplified testing.

I'm quite worried that this change will decrease maintainability (while it removes a lot of code duplicates, it also complicates tests by some - as we now need to mock both buy and selling even if we're not interrested in one side).

Reply:
Yeah I've been thinking this PR might be unnecessary. It didn't really add as much benefit as I was expecting either, I think that I only call it twice within the actual program. I'll close this PR

Reply:
I honestly think closing the PR is a bit premature ...

I'm worried doesn't mean i'm totally against it - or against most parts - more that i need to think about it more.

What i really like is the reduction of duplicate code by moving it to create_order.
while it wouldn't give you benefits for margin - there's other benefits to be had for sure.